server/src/event/writer/mem_writer.rs

```diff
@@ -83,15 +85,20 @@ fn concat_records(schema: &Arc<Schema>, record: &[RecordBatch]) -> RecordBatch {
 }
 
 #[derive(Debug, Default)]
-struct MutableBuffer<const N: usize> {
+struct MutableBuffer {
     pub inner: Vec<RecordBatch>,
     pub rows: usize,
 }
 
-impl<const N: usize> MutableBuffer<N> {
+impl MutableBuffer {
     fn push(&mut self, rb: RecordBatch) -> Option<Vec<RecordBatch>> {
-        if self.rows + rb.num_rows() >= N {
-            let left = N - self.rows;
+        let buf_size = std::env::var(BUF_SIZE)
```

Comment:
we also add a validation - i.e. use the standard route like other env vars (via clap)

Reply:
If we use get the value with clap we would need to add it to the `Server` struct. And will appear in cli options. Is that the behavior we want?

Reply:
I was thinking we should avoid putting this in the help etc, but the pattern of ad hoc env vars seems even more dangerous. So let's stick to Clap

Reply:
Ok